docs/grammar.md

```diff
@@ -59,26 +59,37 @@ Terminals are used to match text into symbols. They can be defined as a combinat
 **Syntax:**
 
 ```html
-<NAME> [. <priority>] : <literals-and-or-terminals>
+<NAME> [. <priority>] : <items-to-match>
 ```
 
-Terminal names must be uppercase.
+Terminal names must be uppercase.  They must start with an underscore (`_`) or a letter (`A` through `Z`), and may be composed of letters, underscores, and digits (`0` through `9`).  Terminal names that start with "_" will not be included in the parse tree, unless the `keep_all_tokens` option is specified.
```

Comment:
Since you add a lot of detail, worth mentioning that terminals that appear in other terminals (i.e. terminal fragments) will be included even if they start with `_`.

Reply:
Good point, will do.